fem/tmop.cpp

```diff
@@ -3151,56 +3168,47 @@ void TMOP_Integrator::AssembleElemVecSurfFit(const FiniteElement &el_x,
    const FiniteElement &el_s = *surf_fit_gf->FESpace()->GetFE(el_id);
 
    const int dof_x = el_x.GetDof(), dim = el_x.GetDim(),
-             dof_s = el_s.GetDof(), nqp = ir_quad.GetNPoints();
+             dof_s = el_s.GetDof();
 
-   Vector surf_fit_gf_e, surf_fit_gf_bar_e;
-   Vector surf_fit_gf_bar_q;
+   Vector sigma_e;
    Array<int> dofs;
    surf_fit_gf->FESpace()->GetElementDofs(el_id, dofs);
-   surf_fit_gf->GetSubVector(dofs, surf_fit_gf_e);
-   surf_fit_gf_bar->GetSubVector(dofs, surf_fit_gf_bar_e);
-   surf_fit_gf_bar->GetValues(el_id, ir_quad, surf_fit_gf_bar_q);
+   int count = 0;
+   for (int s = 0; s < dofs.Size(); s++)
+   {
+      count += ((*surf_fit_marker)[dofs[s]]) ? 1 : 0;
+   }
+   if (count == 0) { return; }
```

Comment:
The `count` check can be moved earlier, and the same for the `GradSurfFit` function.